src/components/toolbar/toolbar.js

```diff
@@ -101,6 +101,12 @@ function mdToolbarDirective($$rAF, $mdConstant, $mdUtil, $mdTheming, $animate) {
 
         attr.$observe('mdScrollShrink', onChangeScrollShrink);
 
+        // If the toolbar has ngShow or ngHide we need to update height immediately as it changed
+        // and not wait for $mdUtil.debounce to happen
+
+        if (attr.ngShow) { scope.$watch(attr.ngShow, updateToolbarHeight); }
```

Comment:
Initially it was `$observe` but than topher came with this jem:
a quick excerpt from here regarding `$observe` verses `$watch` and why we need `$watch` here: http://stackoverflow.com/questions/14876112/angularjs-difference-between-the-observe-and-watch-methods

> (If you try attrs.$observe('attr1') you'll get the string myModel.some_prop, which is probably not what you want.)

The `$observe` was returning the string ‘shouldShowToolbar’, not the value of that expression. And it was only firing once because the expression itself didn’t change.